src/VisualStudio/Core/Def/Implementation/PreviewPane/PreviewPane.xaml.cs

```diff
@@ -108,40 +109,87 @@ private FrameworkElement CreatePreviewElement(IReadOnlyList<object> previewItems
 
             var grid = new Grid();
 
-            for (var i = 0; i < previewItems.Count; i++)
+            foreach (var previewItem in previewItems)
             {
-                var previewItem = previewItems[i];
+                var previewElement = GetPreviewElement(previewItem);
 
-                FrameworkElement previewElement = null;
-                if (previewItem is IWpfDifferenceViewer)
+                // no preview element
+                if (previewElement == null)
                 {
-                    _previewDiffViewer = (IWpfDifferenceViewer)previewItem;
-                    previewElement = _previewDiffViewer.VisualElement;
-                    PreviewDockPanel.Background = _previewDiffViewer.InlineView.Background;
+                    continue;
                 }
-                else if (previewItem is string)
+
+                // the very first preview
+                if (grid.RowDefinitions.Count == 0)
```

Comment:
things like GetPreviewElement can return null but no null checking., things rely on index when first element can be null, return children.count == 0 ? children[0] or returning children[0] without getting it out from visual tree is already wrong kind of stuff.